qdrant-landing/content/documentation/concepts/storage.md

```diff
@@ -437,30 +437,34 @@ client.CreateCollection(context.Background(), &qdrant.CreateCollection{
+  - Uses disk and [RocksDB](https://rocksdb.org/) for persistence.
+  - Fast access but requires a lot of RAM, especially for large payloads like text or images.
 
-In the case of large payload values, it might be better to use OnDisk payload storage.
-This type of storage will read and write payload directly to RocksDB, so it won't require any significant amount of RAM to store.
-The downside, however, is the access latency.
-If you need to query vectors with some payload-based conditions - checking values stored on disk might take too much time.
-In this scenario, we recommend creating a payload index for each field used in filtering conditions to avoid disk access.
-Once you create the field index, Qdrant will preserve all values of the indexed field in RAM regardless of the payload storage type.
+- **On-Disk Storage**:
+  - Reads and writes payloads directly to [Gridstore](/articles/gridstore-key-value-storage/).
+  - Requires less RAM but has higher access latency.
+  - If querying vectors with payload-based conditions, create a payload index for each field to avoid disk access. Indexed fields are kept in RAM.
 
-You can specify the desired type of payload storage with [configuration file](/documentation/guides/configuration/) or with collection parameter `on_disk_payload` during [creation](/documentation/concepts/collections/#create-collection) of the collection.
+You can choose the type of payload storage in the [configuration file](/documentation/guides/configuration/) or by setting the `on_disk_payload` parameter when [creating](/documentation/concepts/collections/#create-collection) a collection.
 
 ## Versioning
 
-To ensure data integrity, Qdrant performs all data changes in 2 stages.
-In the first step, the data is written to the Write-ahead-log(WAL), which orders all operations and assigns them a sequential number.
+Qdrant ensures data integrity through a two-stage process:
+
+1. **Write-Ahead Log (WAL)**:
+   - All data changes are first written to the WAL.
+   - The WAL orders operations and assigns them a sequential number.
+   - Changes in the WAL are safe from power loss.
+
+2. **Segment Updates**:
+   - Changes are then applied to segments.
+   - Each segment keeps the latest version of changes and the version of each point.
+   - If a new change has a lower sequential number than the current version, it is ignored.
 
-Once a change has been added to the WAL, it will not be lost even if a power loss occurs.
-Then the changes go into the segments.
-Each segment stores the last version of the change applied to it as well as the version of each individual point.
-If the new change has a sequential number less than the current version of the point, the updater will ignore the change.
-This mechanism allows Qdrant to safely and efficiently restore the storage from the WAL in case of an abnormal shutdown.
+This process allows Qdrant to restore storage safely from the WAL in case of an unexpected shutdown.
```

Comment:
```suggestion
This process allows Qdrant to restore storage safely from the WAL in case of an unexpected shutdown, by replaying everything from the WAL.
```